src/OpenTelemetry/BaseExporter.cs

```diff
@@ -132,5 +138,13 @@ protected virtual bool OnShutdown(int timeoutMilliseconds)
         protected virtual void Dispose(bool disposing)
         {
         }
+
+        /// <summary>
+        /// Called when the parent <see cref="TracerProvider"/> is set on the exporter.
+        /// </summary>
+        /// <param name="tracerProvider"><see cref="TracerProvider"/>.</param>
+        protected virtual void OnTracerProviderSet(TracerProvider tracerProvider)
```

Comment:
Looks like this will be a public API and we expect 3rd party exporters to use?
It is a bit weird to have `OnTracerProviderSet` on a generic `BaseExporter<T>` (e.g. in the future we might add `OnLoggerProviderSet`, but I guess 50% of that problem was introduced by me 😄.

Reply:
Ya I agree. Don't totally love it. The generics make it tricky. I tried doing something like `BaseExporter<TProvider, TItem>` but that got really messy. I could revist and try to make it work? [Option 3](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1461) was using an interface to make it more opt-in.